docs/proposals/configuration.md

```diff
@@ -0,0 +1,353 @@
+	Cloudhub         *CloudHubConfig         `yaml:"cloudHub" json:"cloudHub"`
+	Modules          *Modules                `yaml:"modules" json:"modules"`
+}
+
+type EdgeControllerConfig struct {
+	NodeUpdateFrequency int32 `yaml:"nodeUpdateFrequency" json:"nodeUpdateFrequency"`
+}
+
+type DeviceControllerConfig struct {
+}
+
+type CloudHubConfig struct {
+	// enable websocket protocol ,default true
+	EnableWebsocket bool `yaml:"enableWebsocket" json:"enableWebsocket"`
+	// open port for websocket server, default 10000
+	WebsocketPort int32 `yaml:"websocketPort" json:"websocketPort"`
+	// enable quic protocol, default false
+	EnableQuic bool `yaml:"enableQuic" json:"enableQuic"`
+	// open prot for quic server, default 10001
+	QuicPort int32 `yaml:"quicPort" json:"quicPort"`
+	// the max incoming stream for quic server, default 10000
+	MaxIncomingStreams int32 `yaml:"maxIncomingStreams" json:"maxIncomingStreams"`
+	// enable unix domain socket protocol, default true
+	EnableUnixSocket bool `yaml:"enableUnixSocket" json:"enableUnixSocket"`
+	// unix domain socket address, default unix:///var/lib/kubeedge/kubeedge.sock
+	UnixSocketAddress string `yaml:"unixSocketAddress" json:"unixSocketAddress"`
+	//default 0.0.0.0
+	Address string `yaml:"address" json:"address"`
+	//default /etc/kubeedge/ca/rootCA.crt
+	TLSCaFile string `yaml:"tlsCaFile" json:"tlsCaFile"`
+	//default /etc/kubeedge/certs/edge.crt
+	TLSCertFile string `yaml:"tlsCertFile" json:"tlsCertFile"`
+	//default /etc/kubeedge/certs/edge.key
+	TLSPrivateKeyFile string `yaml:"tlsPrivateKeyFile" json:"tlsPrivateKeyFile"`
+	//default 30
+	KeepaliveInterval int32 `yaml:"keepaliveInterval" json:"keepaliveInterval"`
+	//default 30
+	WriteTimeout int32 `yaml:"writeTimeout" json:"writeTimeout"`
+	//default 10
+	NodeLimit int32 `yaml:"nodeLimit" json:"nodeLimit"`
+}
+
+type KubeConfig struct {
+	// kube-apiserver address (such as:http://localhost:8080)
+	Master string `yaml:"master" json:"master"`
+	// Enter path to kubeconfig file to enable https connection to k8s apiserver, if master and kubeconfig are both set, master will override any value in kubeconfig.
+	// default "/root/.kube/config"
+	Kubeconfig string `yaml:"kubeconfig" json:"kubeconfig"`
+}
+
+type Modules struct {
+	//default devicecontroller, edgecontroller, cloudhub
+	Enabled []string `yaml:"enabled" json:"enabled"`
+}
+
+```
+
+#### default config file
+
+```yaml
+kube:
+  master: ""
+  kubeconfig: /root/.kube/config
+edgeController:
+  nodeUpdateFrequency: 10
+cloudHub:
+  enableWebsocket: true
+  websocketPort: 10000
+  enableQuic: false
+  quicPort: 10001
+  maxIncomingStreams: 10000
+  enableUnixSocket: true
+  unixSocketAddress: unix:///var/lib/kubeedge/kubeedge.sock
+  address: 0.0.0.0
+  tlsCaFile: /etc/kubeedge/ca/rootCA.crt
+  tlsCertFile: /etc/kubeedge/certs/edge.crt
+  tlsPrivateKeyFile: /etc/kubeedge/certs/edge.key
+  keepaliveInterval: 30
+  writeTimeout: 30
+  nodeLimit: 10
+modules:
+  enabled:
```

Comment:
Is this going to be added or the feature we already have?

Reply:
@chendave , it's actually the content from modules.yaml